src/linux/system.rs

```diff
@@ -776,6 +789,41 @@ fn get_system_info(info: &str) -> Option<String> {
     None
 }
 
+#[cfg(target_os = "android")]
+fn get_system_info(info: InfoType) -> Option<String> {
+    use libc::c_int;
+
+    // https://android.googlesource.com/platform/bionic/+/refs/heads/master/libc/include/sys/system_properties.h#41
+    const PROP_VALUE_MAX: usize = 92;
+
+    extern "C" {
+        fn __system_property_get(name: *const c_char, value: *mut c_char) -> c_int;
+    }
+
+    // https://android.googlesource.com/platform/frameworks/base/+/refs/heads/master/core/java/android/os/Build.java#58
+    let name: &'static [u8] = match info {
```

Comment:
I don't think you need the `: &'static [u8]` part.

Reply:
Rust can't figure out how to unify the type otherwise since they are different lengths:
```
error[E0308]: `match` arms have incompatible types
   --> src\linux\system.rs:806:32
    |
804 |       let name = match info {
    |  ________________-
805 | |         InfoType::Name => b"ro.product.model\0",
    | |                           --------------------- this is found to be of type `&[u8; 17]`
806 | |         InfoType::OsVersion => b"ro.build.version.release\0",
    | |                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected an array with a fixed size of 17 elements, found one with 25 elements
807 | |     };
    | |_____- `match` arms have incompatible types
    |
    = note:   expected type `&[u8; 17]`
            found reference `&'static [u8; 25]`
```

Reply:
Oh. That'd bad. Well, too bad then, thanks for the explanation!